src/lib/math/pcurves/pcurves_impl/pcurves_wrap.h

```diff
@@ -253,7 +253,12 @@ class PrimeOrderCurveImpl final : public PrimeOrderCurve {
                                    std::span<const uint8_t> input,
                                    std::span<const uint8_t> domain_sep) const override {
          if constexpr(C::ValidForSswuHash) {
+#if defined(BOTAN_HAS_XMD)
             return stash(hash_to_curve_sswu<C, false>(hash, input, domain_sep));
+#else
+            BOTAN_UNUSED(hash, input, domain_sep);
+            throw Not_Implemented("Hash to curve not available due to missing XMD");
+#endif
          } else {
             throw Not_Implemented("Hash to curve is not implemented for this curve");
          }
```

Comment:
`if constexpr` and then additionally guarding with `#if defined()` makes me cringe. 😞
It would be amazing if `C::ValidForSswuHash` would actually _depend on_ `BOTAN_HAS_XMD`.